api/src/main/java/org/apache/iceberg/actions/ActionsProvider.java

```diff
@@ -40,6 +40,13 @@ default MigrateTable migrateTable(String tableIdent) {
     throw new UnsupportedOperationException(this.getClass().getName() + " does not implement migrateTable");
   }
 
+  /**
+   * Instantiates an action to migrate an existing table to Iceberg.
+   */
+  default MigrateTable migrateTable(String tableIdent, int parallelism) {
```

Comment:
Instead of extending the base call here I think it makes sense to extend MigrateTable with another functional method like
```java
/**
* @numReaders the number of concurrent file read operations to use per partition
* @return this for method chaining
**/
MigrateTable withParallelReads(int numReaders) {
  // Set some internal property here that is passed through to the list utility method
  return this
}

```

Reply:
We would also need the same for `SnapshotTable`